File: dataset/output/OPENAI_gpt-4o/Task58_OPENAI_gpt-4o.cpp
#include <iostream>
#include <algorithm>

bool XO(const std::string &str) {
    std::string s = str;
    std::transform(s.begin(), s.end(), s.begin(), ::tolower);
    int xCount = std::count(s.begin(), s.end(), 'x');
    int oCount = std::count(s.begin(), s.end(), 'o');
    return xCount == oCount;
}

int main() {
    std::cout << std::boolalpha;
    std::cout << XO("ooxx") << std::endl;  // True
    std::cout << XO("xooxx") << std::endl;  // False
    std::cout << XO("ooxXm") << std::endl;  // True
    std::cout << XO("zpzpzpp") << std::endl;  // True
    std::cout << XO("zzoo") << std::endl;  // False
  
    return 0;
}

File: dataset/output/OPENAI_gpt-4o/Task58_OPENAI_gpt-4o.java
public class Main {
    public static void main(String[] args) {
        System.out.println(XO("ooxx"));
        System.out.println(XO("xooxx"));
        System.out.println(XO("ooxXm"));
        System.out.println(XO("zpzpzpp"));
        System.out.println(XO("zzoo"));
    }

    public static boolean XO(String str) {
        long countX = str.chars().filter(ch -> ch == 'x' || ch == 'X').count();
        long countO = str.chars().filter(ch -> ch == 'o' || ch == 'O').count();
        return countX == countO;
    }
}